daisy/persistence/mongodb_graph_provider.py

```diff
@@ -160,6 +165,11 @@ def read_nodes(self, roi, attr_filter=None, read_attrs=None):
             read_attrs (``list`` of ``string``):
 
                 Attributes to return. Others will be ignored
+
```

Comment:
Can we have a brief description of the expected behavior and fault conditions here? I feel like "left join" is not quite an accurate definition in this case (and even if it is, would help unfamiliar users).